### PR TITLE
Issue/1918

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #1918 

## How Has This Been Tested?
I've added a setting for `Default User Role` in `Advanced` tab of Give Settings and the same user role is reflected when donor performs user registration using donation form.

## Screenshots (jpeg or gifs if applicable):
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/settings/class-settings-advanced.php [hunk 1]
```diff
@@ -64,6 +64,14 @@ public function get_settings() {
 							)
 						),
 						array(
+							'name'    => esc_html__( 'Default User Role', 'give' ),
```
Comment (DevinWalker) [inline includes/admin/settings/class-settings-advanced.php]: You don't have to use `esc_html__()` here. Please just use `__()`

File: includes/actions.php [hunk 2]
```diff
@@ -247,4 +247,22 @@ function give_set_donation_levels_max_min_amount( $form_id ) {
...
+ *
+ * @param array $user_data WP User object.
+ *
+ * @since 1.8.13
+ *
+ * @return mixed
+ */
+function give_assign_user_role( $user_data ) {
+
+	$user_data['role'] = give_get_option( 'donor_default_user_role' );
```
Comment (DevinWalker) [inline includes/actions.php]: this needs to have `subscriber` set as default if the option is not set.

**Comment by ravinderk** [comment]
@mehul0810 Please check my commit in this Pr and make sure you will apply them

**Comment by mehul0810** [comment]
@ravinderk I've checked your commit and will make sure i will implement them.

**Comment by DevinWalker** [comment]
I just tested the upgrade routine on a large data set and it worked nicely. 👍

**Pull request merged**